src/main/java/cloud/klasse/backendbusiness/classroom/ClassroomService.java

```diff
@@ -0,0 +1,92 @@
+package cloud.klasse.backendbusiness.classroom;
+
+import cloud.klasse.backendbusiness.student.CreateStudentModel;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+/**
+ * Classroom service to manage an classroom.
+ *
+ * <p>This class injects the classrom repository.</p>
+ *
+ * @since 0.0.1
+ *
+ * @see Service
+ * @see RequiredArgsConstructor
+ * @see Slf4j
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ClassroomService {
+
+    private final ClassroomRepository classroomRepository;
+
+
+    /**
+     * Create an classroom with the given create classroom model {@link CreateClassroomModel}.
+     *
+     * @param createClassroomModel create classroom model
+     * @return the created classroom
+     *
+     * @since 0.0.1
+     */
+    public Classroom createClassroom(CreateClassroomModel createClassroomModel) {
```

Comment:
The teacher, who is making the request should be specified as teacher here. By auth. Token maybe?

Reply:
You can access the authenticated user like this:
```java
String username = Objects.toString(SecurityContextHolder.getContext().getAuthorization().getPrincipal());
```